test/puppetlabs/http/client/async_plaintext_test.clj

```diff
@@ -452,3 +455,98 @@
                 (let [response @(common/get client url {:as :text})]
                   (is (= 200 (:status response)))
                   (is (= "es-ES" (:body response))))))))))))
+
+(deftest client-route-and-total-limits
+  (testing "client limits 2 requests per route by default"
+    (let [actual-count (atom 0)
+          countdown (CountDownLatch. 3)
+          fake-app (fn [_]
+                     (swap! actual-count inc)
+                     (.countDown countdown)
+                     (.await countdown)
+                     {:status 200
+                      :body "Hello, World!"})]
+      (testlogging/with-test-logging
+        (testwebserver/with-test-webserver fake-app port
+          (let [url (str "http://localhost:" port "/hello")]
+            (testing "clojure persistent async client"
+              (with-open [client (async/create-client {})]
+                (dotimes [n 10] (future (common/get client url {:as :text})))
+                (is (= false (.await countdown 5 TimeUnit/SECONDS)))
+                (is (= 2 @actual-count)))))))))
+
+  (testing "client limits 0 requests per route by 2"
```

Comment:
It seems like the same body as the previous test but that we pass `:max-connections-per-route 0`. It's good to see that the client doesn't blow up, but I'm not grokking the actual affects in the assertions.

Reply:
Oooohhhh, I just double checked the docs you added and I see now

> If 0 is specified, it acts as the default.

Future me would appreciate a slightly more verbose test description (like "passing 0 selects default behavior (a limit of 2)"), else I'm going to have to double check the docs every time I come back to this test.